src/transformState.js

```diff
@@ -5,7 +5,32 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const removeKeys of action.keysToRemove) {
+          if (state.hasOwnProperty(removeKeys)) {
+            delete state[removeKeys];
+          }
+        }
+        break;
+
+      case 'clear':
+        for (const removeProperties in state) {
+          delete state[removeProperties];
+        }
+        break;
+
+      default:
+        return 'Enter an existing type';
```

Comment:
```suggestion
        throw new Error('Enter an existing type');
```

I mean smth like this 😉